jenkins-scripts/docker/lib/_generic_linux_compilation_build.sh.bash

```diff
@@ -102,7 +102,7 @@ else
   echo "Requested: no test run"
 fi
 
-if $GENERIC_ENABLE_CPPCHECK; then
+if [ $GENERIC_ENABLE_CPPCHECK ] && [ -f tools/code_check.sh ]; then
```

Comment:
I think the path is incorrect. I see the following console output in the CI job that should have detected a `tools/code_check.sh` script:

~~~
+ '[' -f tools/code_check.sh ']'
+ echo 'Requested: no ccpcheck run'
~~~

it probably needs to be `[ -f $WORKSPACE/${SOFTWARE_DIR}/tools/code_check.sh ]`